### DISCUSSION TITLE
Our Open Philosophy

### DISCUSSION THREAD

**Discussion by fuksman** [anyproto/contributors#1] [Announcements]
> By opening our source code, we ensure that our users have complete autonomy and independence from the Any Association.

### Trust our code, not our words
We believe that any software should support fundamental digital freedoms. With the rise of cryptography and computer systems, it is now possible to guarantee these freedoms in the world of bits: privacy of thoughts, freedom of speech, right to authorship, and autonomy from software providers. These rights can be encoded into the code, which when open, can be freely verified by anyone. This way, trust among users and developers can be established.

This is our way. By opening our source code, we ensure that our users have complete autonomy and independence from the Any Association. They retain the ability to analyze, compile, and run each software component on their personal machines without relying on external parties. This guarantees uninterrupted access to the tools and data they generate and store, shielding them from any potential restrictions.

### Cooperative Open Source
There are risks associated with opening the code, examples are instances where large corporations have taken advantage of open-source projects like MongoDB and Elastic, reaping benefits without contributing back. Consequently, protective licenses have emerged, obligating cloud service providers to disclose modifications they made and the source code of their tooling. To maintain a delicate balance between our dedication to open source and the idea of self-sustainability, we have developed the Cooperative Open Source philosophy.

While our core solutions, the infrastructure protocol [`any-sync`](https://github.com/anyproto/any-sync), and the data protocol [`any-block`](https://github.com/anyproto/any-block), are released as open source under the permissive MIT license, we distribute the remaining layers, including the middleware library [`anytype-heart`](https://github.com/anyproto/anytype-heart), and applications like [`anytype-ts`](https://github.com/anyproto/anytype-ts), [`anytype-swift`](https://github.com/anyproto/anytype-swift), and [`anytype-kotlin`](https://github.com/anyproto/anytype-kotlin), under the Any Source Available License. This license grants individuals the freedom to review, modify, and utilize the code for personal, academic, scientific, research, and development purposes. However, for commercial use, consent from the Any Association is required.

This way, we guarantee everyone the right to use, modify, and distribute the data exchange protocol and the data format, ensuring that anyone is free to create any application on top of them. We guarantee free, non-commercial usage of the software and full transparency of the code. However, considering the substantial R&D resources required for the application layer, we believe that businesses and networks utilizing our software for commercial purposes should contribute towards its ongoing development, allowing maintainers to support and enhance the platform.

The Any Association, based in Zug, Switzerland, is an organization that will govern the rights to use the software and will provide an opportunity for other significant contributors to join a sort of digital cooperative and become the governors of the software as well. This empowers significant contributors to co-decide the next steps of product development and protects them from rivals’ abuse.

### Co-creation
We see our products as an open ecosystem where the community can contribute in various ways, not only by committing to the core source code. Contributors can develop integrations, design themes, create use cases for the in-app library, or provide documentation and translations.

Documentation and translations created by the Any team and contributors will be in the public domain. As for use cases, integrations, and themes created by the community, the choice of license will be left up to their creators.

We believe this approach to open source could be a sustainable way to deliver the best-in-class software while supporting fundamental digital freedoms in the world of bits.

**Comment by yannicka** [comment]
Anytype is a very great project, however you declare in each of your communications (website, forum, Product Hunt, here on GitHub...) that it is an open source project whereas it is not the case as seen here.

Personally I find it disappointing because the project has been presented for years as open source, but in the end it's not. Of course you have no accountability, and you have the right to change your mind, but it would be good not to present the project as open source as long as it does not respect the rules of the Open Source Initiative or the Free Software Foundation.

**Reply by hit-lacus** [reply]
Maintaining an open source project is not an easy and timely rewarding task.

**Reply by pcjmfranken** [reply]
The license under which AnyType has chosen to publish their source code contains certain clauses that preclude AnyType from being considered open source software. The widespread usage of this term to describe themselves is therefore quite misleading.

I'd highly appreciate it if AnyType would make this right, either by changing to a proper OSS license or by removing all the misleading instances of this term from their copy.

Check out the Open Source Initiative website for further details: https://opensource.org/osd/

**Reply by fuksman** [reply]
@yannicka, @pcjmfranken, thank you for your attention.

The MIT license is used for the foundational part of the project: our any-sync and any-block protocols, which is why we consider ourselves open source. Our client applications are, indeed, licensed under a source-available license, which is not OSI approved. Just to make it clear: a source-available license allows everyone to use, modify, and redistribute the software for non-commercial purposes, whether in personal use cases or within an organization. This license is not considered open-source from an OSI definition perspective because certain commercial uses will require specific permission or license. 

We tried to address our choice and reasoning behind this in this publication, but obviously, we could do better. We didn't mean to mislead anyone, so we'll update our external communication to make it more accurate and to not overuse the term "open source". We apologize if the current wording disappointed you. 
We will also answer all the popular questions about the license, so feel free to ask them here or in separate topics.

**Reply by pcjmfranken** [reply]
Just to be clear, the scope of my criticism was limited to AnyType's nonconforming employment of well-established "open source" terminology. No purposeful insincerity was ever suspected.

Beyond that I am in full support of your core tenets regarding public development, appreciate and respect both your full right to publish your own work under your own terms as well as the specific limitations end up being imposed, and, speaking as someone who has been following AnyType's developments all the way from way back when, you have always done things differently but always ended up back on track - just like it looks is the case again here.

Cheers!

**Comment by suntorytimed** [comment]
One issue regarding the Any Source Available License v1.0. As already stated in the discussion above it's not an OSI compliant open source license. But furthermore it would hinder me from creating and distributing containerised and packaged versions of the software, even if I don't intend to earn any money or have only non-commercial intends with those packages and containers.

> Any Association grants you ("Licensee") a license to use, modify, and redistribute the Software, but only (a) for Non-Commercial Use, or (b) for Commercial Use in Allowed Networks.

In case I would package the anytype-ts client for openSUSE to offer it as an RPM in the official repositories (or in a development repository, as non OSI compliant software wouldn't be accepted into Leap or Tumbleweed), I would possibly break the license agreement, as I can't ensure that I redistribute the software only for Non-Commercial Use or Commercial Use in Allowed Networks. Same is true if I would build a container for any-heart and push it to Docker Hub or a GitHub repository.

**Reply by fuksman** [reply]
Hi @suntorytimed,

The ASAL doesn't restrict you from the actions you've described. When you create a package based on source code from the repositories, it will contain a default network configuration. Users who download and use this software can either use it offline or only with Allowed Networks, which fully complies with the terms.
`anytype-heart` is a middleware library, and it is included in built client apps. Therefore, I don't see any reason to containerize it separately. However, even if you choose to containerize it, and do it without replacing the default network configuration, everything complies with the terms of the license.
Another case, even if you replace the network config (i.e., for your own self-hosted version) and distribute this version, let's say, among your family members, without charging them for usage, it is also totally fine from the terms of the ASAL perspective.

Thank you for the new question. I'll update our [FAQ](https://legal.any.coop/).

**Reply by suntorytimed** [reply]
Hi @fuksman ,

thanks for the clarification. In that case it makes it easier for me to look into packaging the clients for openSUSE and maybe offer the server components for self-hosting as an easy to install RPM. Especially the part with the network configuration wasn't clear to me, as this will nullify my concerns regarding commercial use of the packages as this would only happen in your allowed network per default.

Thanks for clarifying this in your FAQ.

**Reply by fuksman** [reply]
Great, and thank you for your willingness to contribute!

**Comment by dphaldes** [comment]
You have a great product, but I hope you reconsider your license. Having a "source open" project means nothing. If you wish to grow the project and build a strong community around it, please consider using a proper open source license.
There are multiple open source licenses that also work in a commercial setting (like various copyleft licenses) and will help you create a meaningful business.

**Reply by fuksman** [reply]
Thank you for the feedback! It's possible that in the future, the association will change ASAL to an OSI-approved license, but not at the current stage. You've probably seen it already, but just to mention: we're collecting answers to different questions about the license in the [FAQ](https://legal.any.coop/), including the one about alternative licenses like AGPL.
